pkg/server/server.go

```diff
@@ -0,0 +1,109 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package server implements a test server for the SideroLink.
+package server
+
+import (
+	"context"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"io"
+	"time"
+
+	"github.com/siderolabs/kms-client/api/kms"
+)
+
+// Server implents gRPC API.
+type Server struct {
+	kms.UnimplementedKMSServiceServer
+
+	getKey func(string) ([]byte, error)
+}
+
+// NewServer initializes new server.
+func NewServer(keyHandler func(nodeUUID string) ([]byte, error)) *Server {
+	return &Server{
+		getKey: keyHandler,
+	}
+}
+
+// Seal encrypts the incoming data.
+func (srv *Server) Seal(ctx context.Context, req *kms.Request) (*kms.Response, error) {
+	time.Sleep(time.Second)
+
+	key, err := srv.getKey(req.NodeUuid)
+	if err != nil {
+		key, err = GetRandomAESKey()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+
+	aesgcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return nil, err
+	}
+
+	nonce := make([]byte, 12)
+	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+		panic(err.Error())
+	}
+
+	encrypted := aesgcm.Seal(nil, nonce, req.Data, nil)
+
+	return &kms.Response{
+		Data: append(nonce, encrypted...),
+	}, nil
+}
+
+// Unseal decrypts the incoming data.
+func (srv *Server) Unseal(ctx context.Context, req *kms.Request) (*kms.Response, error) {
+	time.Sleep(time.Second)
+
+	key, err := srv.getKey(req.NodeUuid)
+	if err != nil {
+		key, err = GetRandomAESKey()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+
+	aesgcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return nil, err
+	}
+
+	decrypted, err := aesgcm.Open(nil, req.Data[:12], req.Data[12:], nil)
```

Comment:
I wonder if the method should return some garbage (random data) if the `Open` fails to prevent brute-forcing.

Reply:
or just return back encrypted string?

Reply:
if we check the length, we can cut away `nonce`, and return random expected 32 bytes for example